lib/validate.js

```diff
@@ -0,0 +1,45 @@
+const Joi = require ('joi'); 
+const MyPromise = require('./promiselib');
+
+
+const schemaRequired = Joi.object().keys({
+    id: Joi.string().required(),
+    title: Joi.string().regex(/^[a-zA-Z0-9\s]+$/).required(),
+    author: Joi.string().regex(/^[a-zA-Z0-9\s]+$/).required(),
+    added: Joi.string(),
+    isDeleted: Joi.boolean()
+});
+const schemaOptional = Joi.object().keys({
+    id: Joi.string(),
+    title: Joi.string().regex(/^[a-zA-Z0-9\s]+$/),
+    author: Joi.string().regex(/^[a-zA-Z0-9\s]+$/),
+    added: Joi.string(),
+    isDeleted: Joi.boolean()    
+});
+
+module.exports.validateReq = (object) => {
```

Comment:
The only difference between these 2 functions is the chosen schema. You can move this logic to one function that gets the schema as an attribute